dlhub_sdk/utils/publish.py

```diff
@@ -0,0 +1,226 @@
+
+    Args:
+        task (dict): the task description.
+        header (str): the authorization header for the Globus Search Writer Lambda
+    """
+    logger.debug("Ingesting servable into Search.")
+
+    # Construct the subject for the search index metadata as the container id
+    iden = "https://dlhub.org/servables/{}".format(task['dlhub']['id'])
+
+    # We need to construct a "gingest" document to submit to Globus Search
+    # for ingestion https://docs.globus.org/api/search/reference/ingest/#gingest
+    # For this, we need a copy of the metadata dict where all values are strings
+    # to ingest to Globus Search
+    ingestable = task
+    d = [convert_dict(ingestable, str)]
+
+    glist = []
+
+    # Get model visibility from metadata; if there is no 'visible_to' entry
+    # in the metadata, set the visibilty to default to ['public']
+    visible_to = task['dlhub'].get('visible_to', ['public'])
+
+    for document in d:
+        gmeta_entry = mdf_toolbox.format_gmeta(document, visible_to, iden)
+        glist.append(gmeta_entry)
+    gingest = mdf_toolbox.format_gmeta(glist)
+
+    logger.info("ingesting to search")
+    logger.info(gingest)
+
+    GLOBUS_SEARCH_WRITER_LAMBDA = "https://7v5g6s33utz4l7jx6dkxuh77mu0cqdhb.lambda-url.us-east-2.on.aws/"
+
+    # POST the gingest document to the GLOBUS_SEARCH_WRITER_LAMBDA
+    # which will ingest it to the search index.
+
+    http_response = requests.post(
+        GLOBUS_SEARCH_WRITER_LAMBDA,
+        headers={"Authorization": header},
+        json={'document_file': gingest})
+    if http_response.status_code != 200:
+        raise Exception(http_response.text)
+    logger.info("Ingestion of {} to DLHub servables complete".format(iden))
+
+
+def register_funcx(task, container_uuid, funcx_client):
+    """Register the function and the container with funcX.
+
+    Parameters
+    ----------
+    task : dict
+        A dict of the task to publish
+    container_uuid:  str
+    funcx_client:  the funcx client to register with
+
+    Returns
+    -------
+    str
+        The funcX function id
+    """
+
+    fxc = funcx_client
+
+    try:
+        description = task['datacite']['descriptions'][0]['description']
+    except:
+        description = f"A container for the DLHub model {task['dlhub']['shorthand_name']}"
+
+    # The Container Service registers the container w/ funcx
+    # Register a function
+    funcx_id = fxc.register_function(dlhub_run, function_name=task['dlhub']['name'],
+                                     container_uuid=container_uuid, description=description, public=True)
+
+    # Whitelist the function on DLHub's endpoint
+    # Production endpoint is '86a47061-f3d9-44f0-90dc-56ddc642c000'
+    # Dev endpoint is '2238617a-8756-4030-a8ab-44ffb1446092'
+    # endpoint_uuid = '86a47061-f3d9-44f0-90dc-56ddc642c000'
+    endpoint_uuid = '2238617a-8756-4030-a8ab-44ffb1446092'
+
+    res = fxc.add_to_whitelist(endpoint_uuid, [funcx_id])
```

Comment:
We should likely just remove the whitelist option. It isn't being used. We could add it back later if needed.